File: src/tb-detection/factory.py
import marker as m

class MarkerFactory:
    @staticmethod
    def make_markers(dict_length, observer):
        marker_list = []
        for i in range(dict_length):
            marker_list.append(m.Marker(i))
            marker_list[i].attach_observer(observer)
            
            if i == dict_length - 1:
                marker_list[i].set_type("camera")
            elif i >= dict_length - 5:
                marker_list[i].set_type("rotator")
            elif i >= dict_length - 9:
                marker_list[i].set_type("slider")
            else:
                marker_list[i].set_type("model")

        return marker_list

File: src/detector/DetectorTest/Factory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DetectorTest
{
    internal class Factory
    {
        public static List<Marker> MakeMarkers(int dictLength, Observer observer)
        {
            List<Marker> markerList = new List<Marker>();
            for (int i = 0; i < dictLength; i++)
            {
                Marker marker = new Marker(i);
                marker.AttachObserver(observer);

                if (i == dictLength - 1)
                {
                    marker.SetType("camera");
                }
                else if (i >= dictLength - 5)
                {
                    marker.SetType("rotator");
                }
                else if (i >= dictLength - 9)
                {
                    marker.SetType("slider");
                }
                else
                {
                    marker.SetType("model");
                }

                markerList.Add(marker);
            }

            return markerList;
        }
    }
}
